boards: mps2_an521: Fix qemu error on Windows

michal-smola opened the pull request:
QEMU for mps2_an521 uses tty and pty which is not available on Windows hosts. Running a sample in QEMU on Windows shows an error. Fix it by enabling tty on Unix hosts only.

butok reviewed (dismissed):
Thank you for QEMU support on Windows.

ithinuel reviewed (commented):
I’m not familiar with QEMU, Do you know what this line does and what makes it unnecessary on Windows?

michal-smola committed:
boards: mps2: Disable qemu host pty on Windows

pty is not available on Windows hosts. Running a sample in Qemu
on Windows shows an error. Fix it by enabling pty on Unix hosts only.

Signed-off-by: Michal Smola <michal.smola@nxp.com>

michal-smola commented:
The line creates a new pseudo-terminal on the host, connects to it, and redirects the virtual serial port to pseudo-terminal Pseudo-terminal is now available on Windows, it causes QEMU error on Windows.
Updated the PR to reflect changes in the new HW model.

ithinuel commented:
I understand, thank you.
Does that also mean that on Windows, QEMU does not print the output from that serial port (and that tests cannot be executed with twister) ?

michal-smola commented:
@ithinuel QEMU prints output from serial on Windows, because QEMU still uses options -chardev stdio,id=con,mux=on -serial chardev:con, to connect serial to standard input and output of the QEMU process.
